[CPCN-177] - Add `Source` column to Content Activity report

eos87 opened the pull request:
### Purpose
This PR introduces a new column to the Content Activity report.

### What has changed
- Added a new column (`source`) to the report both UI and CSV export
- Removed the usage of `lodash.get` from the `ContentActivity.tsx` component
- Completed typings for the properties of the component
- Remove the usage of `PropTypes` in favour of Typescript interfaces
- Little code clean up 

### Steps to test
- Checkout the branch
- Run the project and go to Reports > Content Activity
- Run the report
- Notice that the column has been added and everything else is working as expected

### Checklist
- [x] This pull request is not adding new forms that use redux
- [x] This pull request is adding missing TypeScript types to modified code segments where it's easy to do so with confidence
- [x] This pull request is replacing `lodash.get` with optional chaining for modified code segments

Thanks for reviewing! 

Resolves: [CPCN-177]

[CPCN-177]: https://sofab.atlassian.net/browse/CPCN-177?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

eos87 committed:
Minor code clean up

CPCN-177

eos87 committed:
Add `Source` column to Content Activity report

This adds the column `source` to the content activity report both in the frontend UI and in the CSV export.

CPCN-177

eos87 committed:
Fix format with black

CPCN-177

eos87 committed:
Implement typings for props and remove `_.get` usage

CPCN-177

petrjasek commented on assets/company-reports/components/ContentActivity.tsx:
```diff
@@ -77,12 +121,12 @@ class ContentActivity extends React.Component<any, any> {
     }
 
-    componentWillReceiveProps(nextProps: any) {
+    UNSAFE_componentWillReceiveProps(nextProps: any) {
```
here it seems like it's doing some derived stated, would be good to [avoid it](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization)

eos87 replied:
yeah, make sense. I'll do the adjustments.

eos87 replied:
done

petrjasek commented on assets/company-reports/components/ContentActivity.tsx:
```diff
@@ -77,12 +121,12 @@ class ContentActivity extends React.Component<any, any> {
     }
 
-    componentWillMount() {
+    UNSAFE_componentWillMount() {
```
I think you can use rather `componentDidMount`

petrjasek reviewed (commented):
looks good, but would rather fix those 2 lifecycle methods

eos87 committed:
Extract report filter into separate component

The intention is to simplify and to improve the `ContentActivity.tsx` component from using UNSAFE cycle methods.

CPCN-177

eos87 committed:
Remove unnecessary state of `ContentActivity.tsx`

CPCN-177

eos87 committed:
Merge remote-tracking branch 'origin/develop' into hg/CPCN-177-content-activity-report-source-column

eos87 committed:
Small UI/UX improvements

CPCN-177

eos87 commented:
Hey @petrjasek, I've made a few changes here: 
- Removed the unnecessary internal state from `ContentActivity.tsx` and used props directly instead 
- Extracted the report filters into a separate component to try to simplify things
- A couple of minor UI/UX improvements discussed earlier today with Fritz and Anna. See screenshots below

Added small label "Select a report" instead of an "empty" dropdown
<img width="450" alt="image" src="https://github.com/superdesk/newsroom-core/assets/124766/545ff3c3-0a5c-4d21-8608-d4dccbc4b8c5">

Fixed alignment and overlapping of the date selector. Also removed the double arrows in the dropdown (those at the beginning)
<img width="450" alt="image" src="https://github.com/superdesk/newsroom-core/assets/124766/10d348ab-ea4c-4bfb-b0e1-b8681801b9d9">

Please take another look when you have some time. Thanks!

petrjasek reviewed (approved):
looks good
